### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by oxytoxin** [luvi-ui/laravel-luvi#5] [Q&A]
Hello All! Great work on this project.

I would like to contribute and would very much appreciate if someone can point a direction to me, like what is your high level thought process when porting a specific component from the shadcn/ui lib.

I also noticed the dashboards on the homepage are fully functional as livewire components. Very dope!

**Comment by anotherfrontendguy** [comment]
Hey, thank you for your kind words! And great to hear you want to contribute 🥳

The current process of porting a component looks something like this:

Step 0:
You need a 'consumer' app where you can test the component you're developing. You can do this in a fresh Laravel installation. We use the Luvi [website repo](https://github.com/luvi-ui/website) for this (because it already has Luvi installed). An example commit illustrating this method can be found here: https://github.com/luvi-ui/website/commit/b978a50db6f86444a1503ef2a3ee3c2588410ac1

A good starting point is to pick a component from the shadcn/ui lib and copy the code from the `code` tab in the docs. Then we try to come as close to this implementation as possible. For example for the Accordion it would start out like this:

```jsx
// original
<Accordion>
  <AccordionItem value="item-1">
    <AccordionTrigger>Is it accessible?</AccordionTrigger>
    <AccordionContent>
      Yes. It adheres to the WAI-ARIA design pattern.
    </AccordionContent>
  </AccordionItem>
</Accordion>
```

```blade
{{-- becomes --}}
<x-accordion>
  <x-accordion.item value="item-1">
    <x-accordion.trigger>Is it accessible?</x-accordion.Trigger>
    <x-accordion.content>
      Yes. It adheres to the WAI-ARIA design pattern.
    </x-accordion.content>
  </x-accordion.item>
</x-accordion>
```

With this we know which components we need to build. The tailwind classes needed to achieve the same visuals can be found in the shadcn source (e.g. [Accordion](https://github.com/shadcn-ui/ui/blob/main/apps/www/registry/new-york/ui/accordion.tsx)). Please note: For now we only support the `New-York` variant, that's important when copying the classes.

Once you have replicated the behavior it's time to move the files into the laravel-luvi package. An example commit for this would be https://github.com/luvi-ui/laravel-luvi/commit/b91c4fad11801fc40b09292c618ea736ba9cdedf. After that you can submit a PR and we'll have a look 🚀

If you have any questions, feel free to ask.

**Reply by thoresuenert** [reply]
We have an unreleased command in the [website repo](https://github.com/luvi-ui/website) to generate all boilerplate files for a new component, we have plans to release it shortly.

**Comment by oxytoxin** [comment]
Thank you for such a detailed response @anotherfrontendguy ! Looking forward to the boilerplate generator @thoresuenert !
